pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -161,7 +183,20 @@ func addFinalizerToGKENetworkParamSet(params *networkv1.GKENetworkParamSet) {
 	params.SetFinalizers(append(gnpFinalizers, GNPFinalizer))
 }
 
-func (c *Controller) syncGKENetworkParamSet(ctx context.Context, key string) error {
+// removeFinalizerInPlace removes a finalizer by mutating params if the finalizer exists
+func removeFinalizerInPlace(params *networkv1.GKENetworkParamSet) {
+	finalizers := params.GetFinalizers()
+	for i, f := range finalizers {
+		if f == GNPFinalizer {
+			finalizers = append(finalizers[:i], finalizers[i+1:]...)
+			break
+		}
+	}
+
+	params.SetFinalizers(finalizers)
+}
+
+func (c *Controller) syncGKENetworkParamSet(ctx context.Context, key string) (err error) {
```

Comment:
don't use a named variable in this case and be explicit on the `return`statements

Reply:
done.